pennylane/measurements/__init__.py

```diff
@@ -156,6 +156,37 @@ def circuit(x):
     In PennyLane we use functions to define measurements (e.g. :func:`counts`). These
     functions will return an instance of the corresponding measurement process
     (e.g. :class:`CountsMP`). This decision is just for design purposes.
+
+Adding your new measurement to PennyLane
+----------------------------------------
+
+If you want PennyLane to natively support your new measurement, you have to make a Pull Request
+creating a file with the name of your measurement (e.g. ``state.py``) and add it in
+``pennylane/measurements/``. This file should contain:
```

Comment:
Should we require users to contribute it to PL? maybe we can just instruct them to implement it in their local version of PL? 

Although it would be nice to outsource the creation of these MPs

Reply:
It's not a requirement, but an encouragement. If the user wants to add its created measurement, and it is relevant to PL, we should tell them how to do it (like for example [this user that wants to add the `purity` measurement](https://github.com/PennyLaneAI/pennylane/issues/3139#issuecomment-1328091079))

Reply:
But I believe @trbromley or @josh146 should have the last word here 😄

Reply:
I think we should re-word this slightly, as there is a difference between:

1. Creating a new measurement that PL natively supports, and
2. Contributing this new measurement back to PL for all users of PL.

(1) can be done simply by teaching developers how to define/create a `MeasurementProcess`. Once defined, the developer can:

- Share it as code with other users
- Create a package/extension for users to import their custom measurements.

If they are interested, they can also choose to contribute this measurement back to PL itself. However, they don't _have_ to contribute it back if their goal is to just have "PennyLane to natively support your new measurement".

Reply:
Changed `If you want PennyLane to natively support your new measurement` to `If you want to add this new measurement to PennyLane such that other users can benefit from it`